gateway/src/apicast/policy/ip_check/client_ip.lua

```diff
@@ -0,0 +1,40 @@
+local ipairs = ipairs
+local re = require('ngx.re')
+
+local _M = {}
+
+local function last_caller_ip()
+  return ngx.var.remote_addr
+end
+
+local function ip_from_x_real_ip_header()
+  return ngx.req.get_headers()['X-Real-IP']
+end
+
+local function ip_from_x_forwarded_for_header()
+  local forwarded_for = ngx.req.get_headers()['X-Forwarded-For']
```

Comment:
There is a posiblity that `get_headers` will be called twice. IMO we should avoid that by passing it from outside.

I've been thinking about this for some time and I think we could at some point extract "request" object to the context. This object would lazy load headers when accessed and cache them.

There is a tradeoff between: loading headers always even when not needed, or twice when using both realip and forwarded for, or complicating the code to load headers only when needed.